File: DATHelper/app/src/main/java/com/kishanprao/datHelper/base/stateMachine/Disconnected.kt
package com.kishanprao.datHelper.base.stateMachine

import android.util.Log

class Disconnected() : AbstractConnectionState() {
    companion object {
        private val TAG = Disconnected::class.java.simpleName
    }

    override fun performAction(action: Action): ConnectionState {
        return when (action) {
            is Action.OnConnect -> {
                Log.v(TAG, "performAction: $action")
                Connected(ConnData(action.serverAddress))
            }
            else -> {
                return super.performAction(action)
            }
        }
    }
}

File: Dragon Android Transfer/Dragon Android Transfer/Base/Wireless/StateMachine/Disconnected.swift
//
//  Disconnected.swift
//  DragonSocket
//
//  Created by Kishan P Rao on 08/11/18.
//  Copyright © 2018 Kishan P Rao. All rights reserved.
//

import Foundation

class Disconnected: AbstractConnectionState {
    override func performAction(_ action: Action) -> ConnectionState {
        let state: ConnectionState
        switch action {
        case Action.OnConnect():
            state = Connected(ConnectionData())
        default:
            state = super.performAction(action)
        }
        return state
    }
}
